docs/pktoken.md

```diff
@@ -0,0 +1,702 @@
+
+```ascii
+BASE64URL(UTF8(JWS Protected Header)) || '.' ||
+BASE64URL(JWS Payload) || '.' ||
+BASE64URL(JWS Signature)
+```
+
+The compact serialization standard does not support a JWS with more than one signature. As our PK Tokens have at least two signatures, we invented a compact serialization format for a JWS with more than one signature:
+
+```ascii
+BASE64URL(JWS Payload) || ':' ||
+BASE64URL(UTF8(JWS Protected Header-OP)) || ':' ||
+BASE64URL(UTF8(JWS Signature-OP)) || ':' ||
+BASE64URL(UTF8(JWS Protected Header-CIC)) || ':' ||
+BASE64URL(UTF8(JWS Signature-CIC)) || ':' ||
+BASE64URL(UTF8(JWS Protected Header-COS)) || ':' ||
+BASE64URL(UTF8(JWS Signature-COS)) || ':' ||
+```
+
+### Refreshed Payload and Signature
+
+If this JWS represents a PK Token, then we may wish to refresh the ID Token. Refreshed ID Tokens typically do not contain the nonce in the initial ID Token. As such, for nonce-commitment PK Tokens, we need to transmit both the initial ID Token that has the nonce-commitment and the refreshed ID Token. To do this, we use the following compact representation:
+
+```ascii
+BASE64URL(JWS Payload) || ':' ||
+BASE64URL(UTF8(JWS Protected Header-OP)) || ':' ||
+BASE64URL(UTF8(JWS Signature-OP)) || ':' ||
+BASE64URL(UTF8(JWS Protected Header-CIC)) || ':' ||
+BASE64URL(UTF8(JWS Signature-CIC)) || ':' ||
+BASE64URL(UTF8(JWS Protected Header-COS)) || ':' ||
+BASE64URL(UTF8(JWS Signature-COS)) || '.' ||
+BASE64URL(JWS Refreshed Payload) || '.' ||
+BASE64URL(UTF8(JWS Refreshed Protected Header-OP)) || ',' ||
+BASE64URL(UTF8(JWS Refreshed Signature-OP)) || '.' ||
+```
+
+# Notes
+
+## GQ Signatures
+
+
+GQ (Guillou and Quisquater) signatures were invented in the paper ['A “Paradoxical” Indentity-Based Signature Scheme Resulting from Zero-Knowledge' (1988).](https://link.springer.com/content/pdf/10.1007/0-387-34799-2_16.pdf) GQ signatures were standardized in  GQ1 in [ISO/IEC 14888-2:2008](https://www.iso.org/standard/44227.html). Our GQ signatures are based on this standard but we have increased the security parameter to 256-bits.
+
+## Our JWS conventions
+
+In this section we discuss how the patterns we use for PK Tokens can be more broadly used for JSON Web Signatures in general.
+
+### The TYP pattern
+
+Because a PK Token always has more two or more signatures we have been forced to think about how to effective organize a JWS (JSON Web Signatures) that two or more signatures. If a JWS has only one signature, then the `iss` (issuer) claim in the payload identifies the party that generated the signature. Once you have two or more signatures, how do you determine which signature matches the issuer? How do you determine the identity of signer that generated each signature? Determining what party generated what signature is important because we need to know where to find which public key to verify which signature.
```

Comment:
*"how to effective**ly** organize"

Reply:
*"that **is** two or more signatures."

Reply:
This sentence is a bit confusing. Is there a way to reword it to make it easier to parse? "Determining what party generated what signature is important because we need to know where to find which public key to verify which signature."

Reply:
Removed the confusing sentence, fixed the other issues